sql/src/main/java/io/crate/operation/reference/information/InformationColumnsExpression.java

```diff
@@ -22,12 +22,35 @@
 package io.crate.operation.reference.information;
 
 import io.crate.metadata.RowContextCollectorExpression;
+import io.crate.types.*;
 import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.collect.MapBuilder;
+
+import java.util.Map;
 
 
 public abstract class InformationColumnsExpression<T>
     extends RowContextCollectorExpression<ColumnContext, T> {
 
+    private final static Integer NUMERIC_PRECISION_RADIX = 2; // Binary
+    private final static Integer DATETIME_PRECISION = 3; // Milliseconds
+
+    /**
+     * For fixed point numbers one bit is used for the sign.
+     * Thus the precision for a ByteType is 8-1 bit with a range
+     * from -128 to 127.
+     *
+     * For floating point numbers please refer to:
+     * https://en.wikipedia.org/wiki/IEEE_floating_point
+     */
+    private static final Map<Integer, Integer> NUMERIC_PRECISION = new MapBuilder<Integer, Integer>()
```

Comment:
maybe rename to `PRECISION_BY_TYPE_ID`, Could also consider using a primitive int-int map